src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -311,30 +320,119 @@ private ExecFileSystem createFuseExecFileSystem(
         storage);
   }
 
-  private @Nullable UserPrincipal getOwner(FileSystem fileSystem) throws ConfigurationException {
+  private @Nullable UserPrincipal getOwner(String name, FileSystem fileSystem)
+      throws ConfigurationException {
     try {
-      return getUser(configs.getWorker().getExecOwner(), fileSystem);
+      return getUser(name, fileSystem);
     } catch (IOException e) {
       ConfigurationException configException =
-          new ConfigurationException("Could not locate exec_owner");
+          new ConfigurationException("Could not locate " + name);
       configException.initCause(e);
       throw configException;
     }
   }
 
+  private static final class Unqueue<T> extends AbstractCollection<T> implements Queue<T> {
+    private final T value;
+
+    Unqueue(T value) {
+      this.value = value;
+    }
+
+    // used methods
+    @Override
+    public T poll() {
+      return value;
+    }
+
+    @Override
+    public boolean add(T o) {
+      checkState(o.equals(value));
+      return true;
+    }
+
+    // unused methods
+    // Queue
+    @Override
+    public T peek() {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public T element() {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public T remove() {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean offer(T o) {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void clear() {
+      throw new UnsupportedOperationException();
+    }
+
+    // Collection
+    @Override
+    public int size() {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+      throw new UnsupportedOperationException();
+    }
+  }
+
   private ExecFileSystem createExecFileSystem(
       InputStreamFactory remoteInputStreamFactory,
       ExecutorService removeDirectoryService,
       ExecutorService accessRecorder,
       ExecutorService fetchService,
-      ContentAddressableStorage storage)
+      ContentAddressableStorage storage,
+      LocalResourceSet resourceSet,
+      String ownerName,
+      List<String> ownerNames)
```

Comment:
suggestion: consider a `Set<String>` or `Collection<String>`?

Reply:
Addressed in previous comment about the owner spectra.